### ISSUE TITLE
nilearn.plotting.find_xyz_cut_coords() returns the wrong coordinate

### ISSUE THREAD

**Issue by banilo** [nilearn/nilearn#1190]
Applying `nilearn.plotting.find_xyz_cut_coords()`on the nifti attached here, a y coordinate is returned that is positive (i.e., in the anterior part of the brain) but is supposed to be negative (in the back of the brain). This bug is not observed with `nilearn.find_cut_slices()`.

[RTPJ-1.nii.zip](https://github.com/nilearn/nilearn/files/397428/RTPJ-1.nii.zip)

**Comment by AlexandreAbraham** [comment]
The code is a bit weird. In some error cases, the fallback si to return a center coordinate. The problem being that this coordinate is not converted to "real world".

```python
    if mask.max() == 0:
        return .5 * np.array(data.shape)
```

**Comment by GaelVaroquaux** [comment]
>     if mask.max() == 0:
>         return .5 * np.array(data.shape)

That's a pretty bad bug!

**Comment by AlexandreAbraham** [comment]
> That's a pretty bad bug!

Also, the file @banilo sent is a very simple one with one component. Because of the thresholding magic activated by default in the code, it is zeroed out and the function fails. I don't understand why this "second threshold" is here but it is definitely not advertised in the description of the function.

**Comment by AlexandreAbraham** [comment]
I'll do a quick PR to refactor the code without changing the behavior of the code.

**Comment by NicolasGensollen** [comment]
Fixed in #2661
